File: src/example/gui/ConsoleUi.java
package org.example.gui;


import org.example.gui.load_menu.menu.LoadMenu;
import org.example.gui.main_menu.menu.MainMenu;
import org.example.presenter.Presenter;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Map;
import java.util.Scanner;

public class ConsoleUI implements View{

    private ArrayList<Map<String,String>> printList;
    private Presenter presenter;
    private final Scanner scanner = new Scanner(System.in);
    private LoadMenu menu;

    public ArrayList<Map<String, String>> getPrintList() {
        return printList;
    }

    public void setPrintList(ArrayList<Map<String, String>> printList) {
        this.printList = printList;
    }

    public LoadMenu getMenu() {
        return menu;
    }

    public Presenter getPresenter() {
        return presenter;
    }

    public Scanner getScanner() {
        return scanner;
    }

    @Override
    public void start() throws IOException {
        print(menu.toString());
        String input = scanner.nextLine();
        menu.getButtons().get(Integer.parseInt(input) - 1).action();

        }


    @Override
    public void print(String string) {
        System.out.println(string);
    }

    @Override
    public void setPresenter(Presenter presenter) {
        this.presenter = presenter;
        this.printList = presenter.getPrintList();
        this.menu = new LoadMenu(this);
    }


}

File: project_tree_lite_python_oop/gui/ConsoleUI.py
from project_tree_lite_python_oop.gui.MenuButton import MenuMain


class View:
    def setPresenter(self, presenter):
        raise "Error, задайте метод"

    def start(self):
        raise "Error, задайте метод"


class ConsoleUi(View):
    def __init__(self):
        self.__printList = None
        self.__menu = MenuMain(self)
        self.__presenter = None

    def getPrintList(self):
        return self.__printList

    def setPrintList(self, printList):
        self.__printList = printList

    def getPresenter(self):
        return self.__presenter

    def start(self):
        self.__presenter.setCommandFromView("load", "", -1)
        self.setPrintList(self.getPresenter().getPrintList())
        flag = False
        while not flag:
            self.__menu.__str__()
            print("exit - выход")
            i_c = input("Введите: ")
            if i_c == "exit":
                flag = True
            else:
                self.__menu.getButtons()[int(i_c) - 1].action()

    def setPresenter(self, presenter):
        self.__presenter = presenter
        self.__printList = presenter.getPrintList()
